src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -783,15 +804,15 @@ TEST_F(ClusterTaskManagerTest, ArgumentEvicted) {
   ASSERT_EQ(leased_workers_.size(), 0);
 
   /* Task argument gets evicted */
-  dependency_manager_.task_ready_ = false;
+  missing_objects_.insert(missing_arg);
   pool_.PushWorker(std::dynamic_pointer_cast<WorkerInterface>(worker));
   task_manager_.ScheduleAndDispatchTasks();
   ASSERT_EQ(dependency_manager_.subscribed_tasks, expected_subscribed_tasks);
   ASSERT_EQ(num_callbacks, 0);
   ASSERT_EQ(leased_workers_.size(), 0);
 
   /* Worker available and arguments available */
-  dependency_manager_.task_ready_ = true;
+  missing_objects_.erase(missing_arg);
```

Comment:
Can we add a test that the pinned object is unpinned when the lease is returned?